src/App.js

```diff
@@ -0,0 +1,61 @@
+import { Home, About, Categories, Flashcards } from './components/views';
+import { Header, Footer, Wrapper, Nav, NavButton } from './components'
+
+import { HashRouter as Router, Route, Switch } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import { useEffect, useState } from 'react';
+
+import { sendData, loadData } from './modules/flashcards';
+
+let isInitial = true;
+
+function App() {
+	const [navIsActive, setNavIsActive] = useState(false)
+	const dispatch = useDispatch()
+	const { categories } = useSelector(state => state.flashcards)
+
+	useEffect(() => {
+		dispatch(loadData())
+	}, [dispatch])
+
+	useEffect(() => {
+		if (isInitial) {
+			isInitial = false;
+			return
+		}
+		dispatch(sendData(categories))
+	//wylączyłam StrictMode, ponieważ wykonywał mi się ten dispatch rownież za pierwszym razem. A chcę tutaj tylko jeden raz nie aktualizować firebase.
+	}, [categories, dispatch])
```

Comment:
Dlatego tak to działa tj. 2x w trybie developerskim ten kod jest uruchamiany, aby właśnie wykluczyć "złą" implementację. 
Zdecydowanie lepiej aktualizować firebase kiedy ktoś potwierdzi przez kliknięcie utworzenie kategorii niż w tym miejscu.